### PR TITLE
Upgrade module pipeline

### PR DESCRIPTION
**Pull request by umarali-nagoor**
Test Results:
=============

Screenshot 2022-11-07 at 5 01 34 PM
Screenshot 2022-11-07 at 5 01 08 PM

Screenshot 2022-11-07 at 5 00 48 PM

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "common-dev-assets"]
+	path = common-dev-assets
+	url = https://github.com/terraform-ibm-modules/common-dev-assets
```
Comment (vburckhardt) [inline .gitmodules:3]: Just add branch = main here (see https://github.com/terraform-ibm-modules/terraform-ibm-transit-gateway/blob/main/.gitmodules)

File: examples/kp-key/outputs.tf [hunk 2]
```diff
@@ -3,11 +3,14 @@
 # Copyright 2021 IBM
 #########################################################################################
```
Comment (vburckhardt) [inline examples/kp-key/outputs.tf:4]: Remove copyright as it's under apache license

File: modules/key-protect/outputs.tf [hunk 3]
```diff
@@ -3,17 +3,22 @@
 # Copyright 2021 IBM
 #########################################################################################
 output "kms_key_crn" {
-  value = ibm_kms_key.key.id
+  value       = ibm_kms_key.key.id
+  description = "kms key crn"
```
Comment (vburckhardt) [inline modules/key-protect/outputs.tf:7]: Could we add full sentences for the descriptions.

File: tests/pr_test.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/cloudinfo"
+	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper"
+)
+
+const basicExampleTerraformDir = "examples/kp-key"
+const resourceGroup = "geretain-test-observability-instances"
```
Comment (vburckhardt) [inline tests/pr_test.go:14]: This needs to be updated to a dedicated resource group for the module - eg: geretain-test-kms.

File: tests/pr_test.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+// There is new feature in terratest-wrapper that will keep Activity Tracker tests on unique regions to avoid conflicts,
+// and using a shared InfoService will allow this feature to work between multiple tests running in parallel.
+func TestMain(m *testing.M) {
+	sharedInfoSvc, _ = cloudinfo.NewCloudInfoServiceFromEnv("TF_VAR_ibmcloud_api_key", cloudinfo.CloudInfoServiceOptions{})
+
+	os.Exit(m.Run())
+}
+
+/*
+*********************************************
```
Comment (vburckhardt) [inline tests/pr_test.go:29]: I'd just remove the comment here

File: tests/pr_test.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	})
+
+	// default options will take care of selecting a good first region, but this test needs a 2nd region for super tenents.
+	// this extra block will select a 2nd region, and due to using a shared Info Service it will select a different region than
+	// the default options selection, or any other parallel test runs.
+	regionOptions := &testhelper.TesthelperTerraformOptions{
+		ExcludeActivityTrackerRegions: true,          // do not select regions that already have an AT
+		CloudInfoService:              sharedInfoSvc, // use pointer to shared info svc to keep track of region seleections
+	}
+	region2, _ := testhelper.GetBestVpcRegionO(options.RequiredEnvironmentVars["TF_VAR_ibmcloud_api_key"], "../common-dev-assets/common-go-assets/cloudinfo-region-vpc-gen2-prefs.yaml", "us-east", *regionOptions)
```
Comment (vburckhardt) [inline tests/pr_test.go:56]: I don't think this is necessary as there is no VPC here

**Review by vburckhardt — COMMENTED** [review]
Hi Umar, thanks for the PR. I put some comments through a first pass

File: modules/key-protect/outputs.tf (hunk 3 above)
Comment (umarali-nagoor) [reply modules/key-protect/outputs.tf]: Added descriptions to both input & output variables at examples & module level.

File: tests/pr_test.go (hunk 4 above)
Comment (umarali-nagoor) [reply tests/pr_test.go]: Pointing to dedicated resource group called geretain-test-kms

File: tests/pr_test.go (hunk 5 above)
Comment (umarali-nagoor) [reply tests/pr_test.go:25]: Removed unnecessary comments.

File: tests/pr_test.go (hunk 6 above)
Comment (umarali-nagoor) [reply tests/pr_test.go]: Removed code related to VPC

**Comment by umarali-nagoor** [comment]
Closing this PR as I have raised below PR by addressing above review comments 

https://github.com/terraform-ibm-modules/terraform-ibm-kms/pull/10

**Pull request closed without merging**